### PR TITLE
[CALCITE-4977] Support Snapshot in RelMdColumnOrigins

### PR DESCRIPTION
**Pull request by lmagic233**
Support Snapshot in RelMdColumnOrigins

https://issues.apache.org/jira/browse/CALCITE-4977

### CODE REVIEW

**Review by asolimando — APPROVED** [review]
LGTM

File: core/src/test/java/org/apache/calcite/test/RelMetadataTest.java [hunk 1]
```diff
@@ -562,6 +562,17 @@ private void checkTwoColumnOrigin(
         true, metadataConfig);
   }
 
+  @ParameterizedTest
+  @MethodSource("getArguments") void testColumnOriginsSnapshot(
+      final MetadataConfig metadataConfig) {
+    checkSingleColumnOrigin(
+        "select productid from products_temporal "
```
Comment (NobiGo) [inline core/src/test/java/org/apache/calcite/test/RelMetadataTest.java:569]: Please reformat the SQL use line breaking

**Review by NobiGo — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/apache/calcite/test/RelMetadataTest.java (hunk 1 above)
Comment (lmagic233) [reply core/src/test/java/org/apache/calcite/test/RelMetadataTest.java]: Done

**Review by NobiGo — APPROVED** [review]
LGTM

**Pull request merged**